codegen/test/chart/templates/rbac.yaml

```diff
@@ -6,43 +6,98 @@
 {{- $painter := $.Values.painter }}
 {{ if $painter.enabled }}
 
+
 ---
 
 kind: ClusterRole
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
-  name: painter-{{ default .Release.Namespace $.Values.painter.namespace }}
+  name: painter-{{ default .Release.Namespace $painter.namespace }}
   labels:
     app: painter
 rules:
 - verbs:
   - GET
+
+---
+
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: painter-{{ default .Release.Namespace $painter.namespace }}
+  labels:
+    app: painter
+subjects:
+- kind: ServiceAccount
+  name: painter
+  namespace: {{ default .Release.Namespace $painter.namespace }}
+roleRef:
+  kind: ClusterRole
+  name: painter-{{ default .Release.Namespace $painter.namespace }}
+  apiGroup: rbac.authorization.k8s.io
+
+{{- $painterNsToResources := dict }}
+{{- $painterNamespacedResources := dict }}
+{{- range $entry := $painter.namespacedRbac }}
+  {{- range $ns := $entry.namespaces }}
+    {{- set $painterNsToResources $ns (concat $entry.resources (get $painterNsToResources $ns | default list) | mustUniq) }}
+  {{- end }}
+  {{- range $resource := $entry.resources }}
+    {{- $_ := set $painterNamespacedResources $resource true }}
+  {{- end }}
+{{- end }}
+
+{{- if not (eq (len $painterNamespacedResources) 1) }}
+---
+
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: painter
+  labels:
+    app: painter
+rules:
+{{- if not (get $painterNamespacedResources secrets) }}
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - GET
+  - LIST
+  - WATCH
+{{- end }}
```

Comment:
Would this work if this condition isn't met and we create a ClusterRole with no rules?

Reply:
yes, you can create an empty clusterRole. I've confirmed it with a quick manual test on my local cluster.

We should avoid that scenario entirely though since we have an [if statement on line 50](https://github.com/solo-io/skv2/pull/481/files/1e0e3f7ff645d9dc622955c1cd47a4c57550f524#diff-66b811f6f6759f3ffc5585c2edac864224c006587e37951d28c7a4fe2be9b906) to prevent that.